tests/phpunit/tests/block-supports/elements.php

```diff
@@ -3,171 +3,306 @@
  * @group block-supports
  *
  * @covers ::wp_render_elements_support
+ * @covers ::wp_render_elements_support_styles
  */
 class Tests_Block_Supports_Elements extends WP_UnitTestCase {
 	/**
-	 * Given a string containing a class prefixed by "wp-elements-" followed by a unique id,
-	 * this function returns a string where the id is one instead of being unique.
-	 *
-	 * @param string $block_content String containing unique id classes.
-	 * @return string String where the unique id classes were replaced with "wp-elements-1".
+	 * @var string|null
 	 */
-	private static function make_unique_id_one( $block_content ) {
-		return preg_replace( '/wp-elements-[a-zA-Z0-9]+/', 'wp-elements-1', $block_content );
+	private $test_block_name;
+
+	public function set_up() {
+		parent::set_up();
+		$this->test_block_name = null;
+	}
+
+	public function tear_down() {
+		WP_Style_Engine_CSS_Rules_Store::remove_all_stores();
+		unregister_block_type( $this->test_block_name );
+		$this->test_block_name = null;
+		parent::tear_down();
 	}
 
 	/**
-	 * Test wp_render_elements_support() with a simple paragraph and link color preset.
-	 * @ticket 54337
+	 * Registers a test block type with the provided color block supports.
+	 *
+	 * @param array $color_settings The color block support settings used for elements.
 	 */
-	public function test_simple_paragraph_link_color() {
-		$result = self::make_unique_id_one(
-			wp_render_elements_support(
-				'<p>Hello <a href="http://www.wordpress.org/">WordPress</a>!</p>',
-				array(
-					'blockName' => 'core/paragraph',
-					'attrs'     => array(
-						'style' => array(
-							'elements' => array(
-								'link' => array(
-									'color' => array(
-										'text' => 'var:preset|color|subtle-background',
-									),
-								),
-							),
-						),
+	public function register_block_with_color_settings( $color_settings ) {
+		$this->test_block_name = 'test/element-block-supports';
+
+		register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 3,
+				'attributes'  => array(
+					'style' => array(
+						'type' => 'object',
 					),
-				)
+				),
+				'supports'    => array(
+					'color' => $color_settings,
+				),
 			)
 		);
-		$this->assertSame(
-			$result,
-			'<p class="wp-elements-1">Hello <a href="http://www.wordpress.org/">WordPress</a>!</p>'
-		);
 	}
```

Comment:
I think this logic should run as part of `set_up()` instead of in the individual tests. So either this method should be removed, or alternatively it should be called in `set_up()`. Also, if we keep the method, it should become `private`.

The block is always registered like that, and `tear_down()` calls the equivalent `unregister_block_type()`, so there should be parity with that.

Reply:
I'm likely missing something here but I'm not following the suggestion. The block type's supports config differs across the data sets used by the test.

While splitting these tests into separate files per function, I've moved the block type registration into the test itself so the per-test configuration might be more obvious.